### PR TITLE
Fix serial migration

### LINKED ISSUE #4267: Hierarchy migration: Higher-level processes without process title
### Problem 
After the the migration of serial publications (hierarchy migration), no process title is assigned to the created process for the toplevel-processes, as for example: periodical, multivolume, ...

### Solution
During the migration of serial publications (hierarchy migration), the process title should be assigned to the toplevel-processes. 

See also: #4268, #4269, #4270, #4271

### Example
hierarchy_processes

### LINKED ISSUE #4268: Hierarchy migration: Higher-level processes with workflow
### Problem 
After the the migration of serial publications (hierarchy migration), a workflow is assigned to the created processes for the toplevel-processes, as for example: periodical, multivolume, ... 

### Solution
During the migration of serial publications (hierarchy migration), no workflow should be assigned to the toplevel-processes. 

See also: #3307, #4267, #4269, #4270, #4271

### Example
hierarchy_processes

### LINKED ISSUE #4270: Hierarchy migration: metadata editor cannot be opened for toplevel-processes
### Problem
After the migration of serial publications (hierarchy migration), the created processes for the toplevel-processes, as for example: periodical, multivolume, ... cannot be opened in the metadata editor. The following error message occurs: 

- `Editing metadaten of process 435,649 locked, metadata file belongs to process 370007!Editing metadaten of process 435,649 locked, metadata file belongs to process 370007!`

The first number is the ID of the toplevel-process, the second number is the ID of one of its subordinate processes. 

### Solution
It should be possible to open the toplevel-processes in the metadata editor. 
Additionally, the first ID could be depicted without comma `435,649` -> `435649`. 

See also: #3307, #4267, #4268, #4269, #4271

### Example
errormessage_open_metadata

### PR DESCRIPTION
**Pull request by Kathrin-Huber**
fixes #4270 
resolves #4268 
fix #4267

### CODE REVIEW

**Comment by andre-hohmann** [comment]
The tests were successful: 
- The superordinate processes can be opened in the metadata editor and the correct id is inserted in the METS-file
- The process title of the superordinate processes are generated correctly
- The superordinate processes have no workflow

File: Kitodo/src/main/java/org/kitodo/production/helper/tasks/HierarchyMigrationTask.java [hunk 1]
```diff
@@ -242,6 +247,33 @@ private static void migrateMetadataFiles(Process process) throws IOException {
         return parentData;
     }
 
+    private void checkTaskAndId(Process parentProcess) throws IOException {
+        URI parentMetadataFilePath = fileService.getMetadataFilePath(parentProcess, true, true);
+        Workpiece workpiece = ServiceManager.getMetsService().loadWorkpiece(parentMetadataFilePath);
+        ImportService.checkTasks(parentProcess, workpiece.getRootElement().getType());
```
Comment (solth) [inline Kitodo/src/main/java/org/kitodo/production/helper/tasks/HierarchyMigrationTask.java:253]: If the method `checkTasks` is required here as well (and not just during import), wouldn't it make sense to move that method from `ImportService` to `WorkflowControllerService` or `ProcessService`?

File: Kitodo/src/test/java/org/kitodo/production/helper/tasks/HierarchyMigrationTaskIT.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        cleanUp();
+    }
+
+    /**
+     * Tests migration of hierarchies.
+     */
+    @Test
+    public void testHierarchyMigration() throws DAOException, ProcessGenerationException, CommandException,
+            DataException, IOException, SAXException, ParserConfigurationException {
+        HierarchyMigrationTask hierarchyMigrationTask = new HierarchyMigrationTask(Arrays.asList(project));
```
Comment (solth) [inline Kitodo/src/test/java/org/kitodo/production/helper/tasks/HierarchyMigrationTaskIT.java:85]:
```suggestion
        HierarchyMigrationTask hierarchyMigrationTask = new HierarchyMigrationTask(Collections.singletonList(project));
```

File: Kitodo/src/test/java/org/kitodo/production/helper/tasks/HierarchyMigrationTaskIT.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.helper.tasks;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
```
Comment (solth) [inline Kitodo/src/test/java/org/kitodo/production/helper/tasks/HierarchyMigrationTaskIT.java:16]:
```suggestion
import java.util.Collections;
```

File: Kitodo/src/test/java/org/kitodo/production/helper/tasks/HierarchyMigrationTaskIT.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            ServiceManager.getProcessService().getById(4).getTasks().isEmpty());
+        Assert.assertEquals("JahrdeDeG_404810993", ServiceManager.getProcessService().getById(4).getTitle());
+        Document document = DocumentBuilderFactory.newInstance().newDocumentBuilder()
+                .parse(new File("src/test/resources/metadata/4/meta.xml"));
+        String documentId = document.getElementsByTagName("mets:metsDocumentID").item(0).getTextContent();
+        Assert.assertEquals("DocumentId not set", "4", documentId);
+    }
+
+    private static void createTestMetaAnchorfile() throws Exception {
+        List<String> lines = Arrays.asList("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n"
```
Comment (solth) [inline Kitodo/src/test/java/org/kitodo/production/helper/tasks/HierarchyMigrationTaskIT.java:97]:
```suggestion
        List<String> lines = Collections.singletonList("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n"
```

File: Kitodo/src/test/java/org/kitodo/production/helper/tasks/HierarchyMigrationTaskIT.java [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+                + "        <mets:div ID=\"LOG_0000\" DMDID=\"DMDLOG_0000\" TYPE=\"MultiVolumeWork\" ORDERLABEL=\"Jahresbericht der Deutschen Gerberschule zu Freiberg in Sachsen\" LABEL=\"Jahresbericht der Deutschen Gerberschule zu Freiberg in Sachsen\">\n"
+                + "            <mets:div ID=\"LOG_0001\" TYPE=\"PeriodicalVolume\">\n"
+                + "                <mets:mptr xlink:href=\"\" LOCTYPE=\"URL\"/>\n" + "            </mets:div>\n"
+                + "        </mets:div>\n" + "    </mets:structMap>\n" + "</mets:mets>\n");
+        File processHome = new File(ConfigCore.getKitodoDataDirectory(), "2");
+        FileUtils.writeLines(new File(processHome, "meta_anchor.xml"), "UTF-8", lines);
+    }
+
+    private static void createTestMetafile() throws Exception {
+        List<String> lines = Arrays.asList("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n"
```
Comment (solth) [inline Kitodo/src/test/java/org/kitodo/production/helper/tasks/HierarchyMigrationTaskIT.java:124]:
```suggestion
        List<String> lines = Collections.singletonList("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n"
```

File: Kitodo/src/test/java/org/kitodo/production/helper/tasks/HierarchyMigrationTaskIT.java [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.helper.tasks;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
```
Comment (solth) [inline Kitodo/src/test/java/org/kitodo/production/helper/tasks/HierarchyMigrationTaskIT.java:16]:
```suggestion
```

**Review by solth — COMMENTED** [review]
You forgot to remove one unused import

**Review by solth — APPROVED** [review]

**Pull request merged**